### ISSUE TITLE
Unable to use a 3rd party library json2typescript

### ISSUE THREAD

**Issue by savanvadalia** [ionic-team/ionic-app-scripts#1038]
#### Short description of the problem:

Unable to use a 3rd party lib [json2typescript](https://www.npmjs.com/package/json2typescript).
It throws below error 
`
Module build failed: Error: ENOENT: no such file or directory, open 'C:\Savan\TestProjects\cutepuppypics\node_modules\json2typescript\index.js'
`
![image](https://user-images.githubusercontent.com/17091755/27070812-d8e6ac70-505d-11e7-8b82-c17964fed594.png)

#### What behavior are you expecting?
I should be able to use this 3rd party library

**Steps to reproduce:**

1. Install the cutepuppypics kick-starter ionic project 
2. Install the json2typescript npm package.
  `npm install json2typescript --save`
3. Open any .ts file in the kick-starter project and import the json2typescript module
  `import { JsonConvert } from "json2typescript";`
4. Add below line of code to any method in the above ts file
   `JsonConvert.debugMode = true;`
5. Above line is needed as it seems to telling the ionic build process that it needs to package the 'json2typescript' lib for the final app bundle. Only importing the module doesn't use the package in the bundling process.
6. Save all the changes and do `ionic serve` it should show the same error I am getting.

**Which @ionic/app-scripts version are you using?**
```
Cordova CLI: 6.5.0
Ionic Framework Version: 3.3.0
Ionic CLI Version: 2.2.1
Ionic App Lib Version: 2.2.0
Ionic App Scripts Version: 1.3.7
ios-deploy version: Not installed
ios-sim version: Not installed
OS: Windows 10
Node Version: v6.9.1
Xcode version: Not installed
```

**Comment by raviada** [comment]
I am running into the same problem. Have you solved it?

**Comment by savanvadalia** [comment]
@raviada the issue I have created is the problem of the lib itself reason being there is no packaged/build version of the lib. As a temporary fix I have just copied the lib file to my project source (there is just one file to work with). 

There is a finished pull request waiting to be merged by the repo owner which will solve this manual copying. See the pull request here https://github.com/dhlab-basel/json2typescript/pull/15.

However, now with my temporary fix I am running into another problem which I think is a ionic-app-scripts issue. Now the build fails where my app uses the lib deserialization functionality when running in production mode `ionic run android -prod` mode only. Surprisingly, it works perfectly fine with `ionic serve` or `ionic run android` (dev mode) or `ionic run android -aot`. 
So, I think ionic-app-scripts optimisation(minifying) script is causing the issue here.

I still haven't investigated enough to report an issue with an example.
